docs/changelog/3_x.rst

```diff
@@ -0,0 +1,517 @@
+          },
+        },
+      },
+    }
+
+It's a super-handy way to quickly explore your data.
+
+
+Nested modules
+--------------
+
+"Yo, dawg. We heard you like modules…" (Sorry. We've been working so hard on
+3.0, `we got way way behind on our memes
+<https://knowyourmeme.com/memes/xzibit-yo-dawg>`_.)
+
+You can now put a module inside another module to let you organize your schema
+in any way that makes sense to you.
+
+.. code-block:: sdl
+
+    module MommaModule {
+      module BabyModule {
+        <schema-declarations>
+      }
+    }
+
+
+``intersect`` and ``except`` operators
+--------------------------------------
+
+Slice and dice your sets in new ways with the ``intersect`` and ``except``
+operators. Use ``intersect`` to find common members between sets.
+
+.. code-block:: edgeql-repl
+
+    db> select {1, 2, 3, 4, 5} intersect {3, 4, 5, 6, 7};
+    {3, 5, 4}
+
+Use ``except`` to find members of the first set that are not in the second.
+
+.. code-block:: edgeql-repl
+
+    db> select {1, 2, 3, 4, 5} except {3, 4, 5, 6, 7};
+    {1, 2}
+
+
+
+Query plan
+----------
+
+Among other improvements, the UI now includes a visual query planner to help
+you tweak performance on your EdgeQL queries. Just drop the ``analyze`` keyword
+in front of your query in the UI's "Query Editor" tab to see the query planner
+in action.
+
+.. image:: images/v3_ui.jpg
+    :width: 100%
+
+Query planning is available in the CLI REPL, also by prepending your query with
+``analyze``.
+
+
+Additional changes
+==================
+
+EdgeQL
+------
+
+* Add :eql:func:`assert` function (:eql:gh:`#5040`)
+
+* Support custom user-defined error messages for access policies
+  (:eql:gh:`#4529`)
+
+  .. code-block:: sdl
+
+      access policy admin_only
+        allow all
+        using (global current_user.is_admin ?? false) {
+          errmessage := 'Only admins may query Users'
+        };
+
+* Support casting a UUID to a type (:eql:gh:`#4469`). This is a handy way to
+  select an object, assuming the type you cast into has an object with the UUID
+  being cast.
+
+  .. code-block:: edgeql-repl
+
+      db> select <Hero><uuid>'01d9cc22-b776-11ed-8bef-73f84c7e91e7';
+      {default::Hero {id: 01d9cc22-b776-11ed-8bef-73f84c7e91e7}}
+
+* Add the :eql:func:`json_object_pack` function to construct JSON from an array
+  of key/value tuples. (:eql:gh:`#4474`)
+
+  .. code-block:: edgeql-repl
+
+      db> select json_object_pack({("hello", <json>"world")});
+      {Json("{\"hello\": \"world\"}")}
+
+* Support tuples as query arguments (:eql:gh:`#4489`)
+
+  .. code-block:: edgeql
+
+      select <tuple<str, bool>>$var;
+      select <optional tuple<str, bool>>$var;
+      select <tuple<name: str, flag: bool>>$var;
+      select <optional tuple<name: str, flag: bool>>$var;
+
+* Implement migration rewrites (:eql:gh:`#4585`)
+
+* Implement schema reset (:eql:gh:`#4714`)
+
+* Support link properties on computed backlinks (:eql:gh`#5227`)
+
+
+CLI
```

Comment:
* `edgedb watch`
* `edgedb migrate --dev-mode`
* `edgedb migration upgrade-check`
  * at some point we should not forget that `upgrade` now executes `upgrade-check` (but not yet)
* `edgedb migration create --squash`
* `edgedb analyze`, `\analyze`, `--expand`
* `edgedb info --get` 2.1 maybe worth mentioning
`--channel=testing, --to-testing` in `instance create`, `project init` and `cli upgrade`
* `\d object xxx` -> `\d xxx` maybe worth mentioning?
* Cloud support?

Reply:
Thank you for this list, @tailhook. I just added `watch` and the development workflow — an abridged version of what I wrote previously for the documentation. I also added mention of the `analyze` commands to the section on query planning, although I don't know what `--expand` is or how to use it. I'll circle back now and pick up the testing channel stuff and the new syntax of the describe object backslash command.

Can you fill me in on these?

* `edgedb migrate --dev-mode`
* `--expand option`

My thinking on Cloud support is that, even though we have already added it, it's not relevant to mention until Cloud launches. Open to hear arguments against that approach.

Reply:
Just to spell out what I need to know:

* `edgedb migrate --dev-mode` isn't documented on `--help` best I can tell. I created a new migration and ran it and there was no output. Can't really tell what it does.
* I'm not sure what command `--expand` is an option for.